packages/client/src/app/pages/forms/edit/builder/tabs/layout-editor/property-editor/form-controls/options-editor/custom-options.tsx

```diff
@@ -0,0 +1,292 @@
+import React, { useEffect, useState } from 'react';
+import {
+  AddOptionButton,
+  CheckboxWrapper,
+  Column,
+  DeleteButton,
+  DeleteIcon,
+  DragButton,
+  DragIcon,
+  H3,
+  Input,
+  PlusIcon,
+  Row,
+  TableAddOption,
+  TableOptions,
+  Wrapper,
+} from '@editor/builder/tabs/layout-editor/property-editor/form-controls/options-editor/custom-options.styles';
+import { CheckboxInput } from '@ff-client/app/components/form-controls/inputs/checkbox-input';
+import type { GenericValue } from '@ff-client/types/properties';
+
+type OptionProp = {
+  label: string;
+  value: string;
+  default: boolean;
+  autofocus?: boolean;
+};
+
+type PayloadProp = {
+  index: number;
+  key: string;
+  value: string | boolean;
+};
+
+export type CustomOptionsProps = {
+  useCustomValues: boolean;
+  options: OptionProp[] | GenericValue[];
+};
+
+type Props = {
+  handle: string;
+  value: CustomOptionsProps | GenericValue;
+  onChange?: (customOptions: CustomOptionsProps) => void;
+};
+
+const CustomOptions: React.FC<Props> = ({ handle, value, onChange }) => {
+  const enterKeyPress = useKeyPress('Enter');
+
+  /**
+   * Adds blank label/value object
+   */
+  const addOption = (): void => {
+    const options = value.options ? [...value.options] : [];
+
+    options.push({
+      label: '',
+      value: '',
+      default: false,
+      autofocus: true,
+    });
+
+    if (onChange) {
+      onChange({
+        ...value,
+        options,
+      });
+    }
+  };
+
+  /**
+   * Filters out the option based on its index
+   * @param optionIndex
+   */
+  const deleteOption = (optionIndex: number): void => {
+    let options = JSON.parse(JSON.stringify(value.options));
+
+    options = options.filter(
+      (option: OptionProp, index: number) => index !== optionIndex
+    );
+
+    if (onChange) {
+      onChange({
+        ...value,
+        options,
+      });
+    }
```

Comment:
```suggestion
```suggestion
    onChange({
      ...value,
      options,
    });
```
```